autoPyTorch/evaluation/utils.py

```diff
@@ -102,3 +103,54 @@ def _predict(self, X: np.ndarray) -> np.ndarray:
             predictions.append(pred.ravel())
 
         return np.asarray(predictions).T
+
+
+class DisableFileOutputParameters(Enum):
+    """
+    Contains literals that can be passed in to `disable_file_output` list.
+    These include:
+
+    + `y_optimization`:
+        do not save the predictions for the optimization set,
+        which would later on be used to build an ensemble. Note that SMAC
+        optimizes a metric evaluated on the optimization set.
+    + `pipeline`:
+        do not save any individual pipeline files
+    + `pipelines`:
+        In case of cross validation, disables saving the joint model of the
+        pipelines fit on each fold.
+    + `y_test`:
+        do not save the predictions for the test set.
+    + `all`:
+        do not save any of the above.
+    """
+    pipeline = 'pipeline'
+    pipelines = 'pipelines'
+    y_optimization = 'y_optimization'
+    y_test = 'y_test'
+    all = 'all'
+
+    @classmethod
+    def check_compatibility(cls, disable_file_output: List) -> None:
+        for item in disable_file_output:
+            if item not in cls.__members__ and not isinstance(item, cls):
+                raise ValueError(f"Expected {item} to be in the members ("
+                                 f"{list(cls.__members__.keys())}) of {cls.__name__}"
+                                 f" or an instance.")
+
+    @staticmethod
+    def check_value_in_iterable(container: Iterable, parameter: "DisableFileOutputParameters") -> bool:
+        """
+        checks if parameter is in the container either as
```

Comment:
I think this method should not be staticmethod, but it should look something like:
```python
def is_in_iterable(self, iter: Iterable) -> bool:
    return self in iter or self.value in iter
```

Btw, I think it is more appropriate to use `is_xxx` when we return `bool` and we should use `check_xxx` when we check if inputs are compatible and if not raise error.

Reply:
yeah definitely makes sense, I'll change this.

Reply:
If subclassing a self made enum class that provides an equality method might make things more intuitive.

Note: I havn't tested this but this is the main idea

```python
class MyEnum(Enum):

    def __eq__(self, other: Any) -> bool:
        if isinstance(other, MyEnum):
            return self == other
        elif isinstance(other, str):
            return self.value = other.value
        else:   
            raise RuntimeError("Some error")

class SubEnum(MyEnum):
    ... inherits __eq__
    

```.

```python
# Dict uses __eq__ in it's __contain__ method
if my_enum_val in my_dict:
    ...
    
# Same goes for list
```